src/test/java/com/sap/sgs/phosphor/fosstars/data/github/GitHubDataFetcherTest.java

```diff
@@ -79,4 +92,349 @@ public void testShouldUpdate() throws IOException {
+    ExecutorService executorService = Executors.newFixedThreadPool(10);
+
+    GitHubProject project = new GitHubProject("test", "project");
+    Path projectDir = projectDir(project.name());
+    try (
+        Repository repository = FileRepositoryBuilder.create(projectDir.resolve(".git").toFile())) {
+      repository.create();
+
+      createRepoTestFile(repository, project, "testFile");
+
+      fetcher.addRepositoryInfoForTesting(project, projectDir);
+
+      int numberOfThreads = 25;
+      CountDownLatch latch = new CountDownLatch(numberOfThreads);
+      for (int i = 0; i < numberOfThreads; i++) {
+        executorService.submit(() -> {
+          try {
+            localRepositoryFor(project);
+          } catch (IOException e) {
+            throw new IllegalStateException("Load repository exception");
+          }
+          latch.countDown();
+        });
+      }
+      latch.await();
+
+      checkLocalRepository(project, 1);
+    } finally {
+      executorService.shutdown();
+    }
+  }
+
+  @Test
+  public void testMultiThreadCleanUpTest() throws IOException, InterruptedException {
+    ExecutorService executorService = Executors.newFixedThreadPool(10);
+
+    GitHubProject project = new GitHubProject("test", "project");
+    Path projectDir = projectDir(project.name());
+    try (
+        Repository repository = FileRepositoryBuilder.create(projectDir.resolve(".git").toFile())) {
+      repository.create();
+
+      createRepoTestFile(repository, project, "testFile");
+
+      fetcher.addRepositoryInfoForTesting(project, projectDir);
+
+      localRepositoryFor(project, 1);
+
+      int numberOfThreads = 25;
+      CountDownLatch latch = new CountDownLatch(numberOfThreads);
+      for (int i = 0; i < numberOfThreads; i++) {
+        executorService.submit(() -> {
+          try {
+            cleanup(project);
+          } catch (IOException e) {
+            throw new IllegalStateException("Clean up exception");
+          }
+          latch.countDown();
+        });
+      }
+      latch.await();
+
+      checkCleanUp(project, 0);
+    } finally {
+      executorService.shutdown();
+    }
+  }
+
+  @Test
+  public void testLocalRepositoriesInfoJson() throws IOException {
+    // Create the first project with first GitHub mock instance.
+    GitHubProject firstProject = new GitHubProject("test", "project1");
+    Path project1Dir = projectDir(firstProject.name());
+    GitHub github = mock(GitHub.class);
+    try (Repository repository =
+        FileRepositoryBuilder.create(project1Dir.resolve(".git").toFile())) {
+      repository.create();
+
+      createRepoTestFile(repository, firstProject, "testFile1");
+
+      fetcher = spy(new TestGitHubDataFetcher(github));
+
+      fetcher.addRepositoryInfoForTesting(firstProject, project1Dir);
+
+      localRepositoryFor(firstProject, 1);
+    }
+
+    // Create the second project with second GitHub mock instance.
```

Comment:
Is this comment relevant? If I understand correctly, the test checks loading local repositories. But this doesn't involve the `GitHub` instance.

Reply:
The entire test is another duplicate. I am testing the `json` file is by calling `loadLocalRepositoriesInfo()` in `getLocalRespositoriesInfoForTesting()`(class `TestGitHubDataFetcher`). Hence, since it is being checked in every other test. It does not need to be verified explicitly here.

Reply:
https://github.com/SAP/fosstars-rating-core/pull/246#discussion_r448773776